interp/decls.mli

```diff
@@ -10,32 +10,77 @@
 
 open Names
 open Libnames
-open Decl_kinds
+
+type theorem_kind =
+  | Theorem
+  | Lemma
+  | Fact
+  | Remark
+  | Property
+  | Proposition
+  | Corollary
+
+type definition_object_kind =
+  | Definition
+  | Coercion
+  | SubClass
+  | CanonicalStructure
+  | Example
+  | Fixpoint
+  | CoFixpoint
+  | Scheme
+  | StructureComponent
+  | IdentityCoercion
+  | Instance
+  | Method
+  | Let
+
+type assumption_object_kind = Definitional | Logical | Conjectural | Context
+
+(* [assumption_kind]
+
+                |  Local      | Global
+   ------------------------------------
+   Definitional |  Variable   | Parameter
+   Logical      |  Hypothesis | Axiom
+
+*)
+
+(** Kinds used in library *)
+
+type logical_kind =
+  | IsPrimitive
+  | IsAssumption of assumption_object_kind
+  | IsDefinition of definition_object_kind
+  | IsProof of theorem_kind
 
 (** This module manages non-kernel informations about declarations. It
     is either non-logical informations or logical informations that
     have no place to be (yet) in the kernel *)
 
 (** Registration and access to the table of variable *)
 
-type variable_data = {
-  path:DirPath.t;
-  opaque:bool;
-  univs:Univ.ContextSet.t;
-  poly:bool;
-  kind:logical_kind;
-}
+type variable_data =
+  { path:DirPath.t
+  ; opaque:bool
+  ; univs:Univ.ContextSet.t
+  ; poly:bool
+  ; kind:logical_kind
+  }
 
 val add_variable_data : variable -> variable_data -> unit
+
+(* Not used *)
 val variable_path : variable -> DirPath.t
```

Comment:
Why keep it then?

Reply:
I though we could address the refactoring of this module in its own PR.